main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.RouteBothShortAndLongNameMissingNotice;
+import org.mobilitydata.gtfsvalidator.notice.RouteShortAndLongNameEqualNotice;
+import org.mobilitydata.gtfsvalidator.notice.RouteShortNameTooLongNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+
+/**
+ * Validates short and long name for a single route.
+ * <p>
+ * Generated notices:
+ * * RouteBothShortAndLongNameMissingNotice
+ * * RouteShortAndLongNameEqualNotice
+ * * RouteShortNameTooLongNotice
+ */
+@GtfsValidator
+public class RouteNameValidator extends SingleEntityValidator<GtfsRoute> {
+    private static final int MAX_SHORT_NAME_LENGTH = 6;
+
+    @Override
+    public void validate(GtfsRoute entity, NoticeContainer noticeContainer) {
+        final boolean hasLongName = entity.hasRouteLongName();
+        final boolean hasShortName = entity.hasRouteShortName();
+
+        if (!hasLongName && !hasShortName) {
+          noticeContainer.addNotice(new RouteBothShortAndLongNameMissingNotice(
+              entity.routeId(), entity.csvRowNumber()));
+        }
+
+        if (hasShortName && hasLongName &&
+            entity.routeShortName().equalsIgnoreCase(entity.routeLongName())) {
```

Comment:
This rule could be expanded from "equals" to "contains" - the [spec best practices](https://gtfs.org/best-practices/#routestxt) say that:
>route_long_name should not contain the route_short_name.

Reply:
Warming though that can be tricky to find a good heuristic. Some long name contain the short one for good reasons in some network. I guess it's ok for a warning is we're doing something basic.

Reply:
This best practice may be hard to check and follow: consider `route_short_name` = 'C' and `route_long_name` = 'Central'. What we have at Google is:

* check that long and short name are different;
* check that long name does not start from the short name followed by ' ', '-' or '('.

The second check is tricky and we can add it later.

Let's start from just comparing long and short names.

Reply:
Good points - ok, let's stick with equals for now.

Reply:
Actually, we already have a six-month-old issue for this that I forgot about 😆  - https://github.com/MobilityData/gtfs-validator/issues/267